core/controllers/suggestion_test.py

```diff
@@ -475,3 +723,218 @@ def test_accept_question_suggestion(self):
+                    self.topic_id, suggestion_id), {
+                        'action': u'reject',
+                        'review_message': u'Rejected!'
+                    }, csrf_token=csrf_token, expected_status_int=400)
+
+        self.assertEqual(
+            response['error'],
+            'This handler allows actions only on suggestions to topics.')
+
+        self.logout()
+
+    def test_suggestion_to_topic_handler_with_invalid_target_id(self):
+        self.login(self.ADMIN_EMAIL)
+
+        response = self.get_html_response(feconf.CREATOR_DASHBOARD_URL)
+        csrf_token = self.get_csrf_token_from_response(response)
+
+        suggestion_to_accept = self.get_json(
+            '%s?author_id=%s' % (
+                feconf.SUGGESTION_LIST_URL_PREFIX,
+                self.author_id))['suggestions'][0]
+
+        self.save_new_topic(
+            'topic_id', self.admin_id, 'Name1', 'Description',
+            [], [], [], [], 1)
+
+        response = self.get_html_response(feconf.CREATOR_DASHBOARD_URL)
+        csrf_token = self.get_csrf_token_from_response(response)
+
+        with self.swap(constants, 'ENABLE_NEW_STRUCTURE_PLAYERS', True):
+            response = self.put_json(
+                '%s/topic/%s/%s' % (
+                    feconf.SUGGESTION_ACTION_URL_PREFIX,
+                    'topic_id', suggestion_to_accept['suggestion_id']),
+                {
+                    'action': u'reject',
+                    'review_message': u'Rejected!'
+                },
+                csrf_token=csrf_token, expected_status_int=400)
+
+        self.assertEqual(
+            response['error'],
+            'The topic id provided does not match the topic id '
+            'present as part of the suggestion_id')
+
+        self.logout()
+
+    def test_suggestion_to_topic_handler_with_invalid_action(self):
+        self.login(self.ADMIN_EMAIL)
+
+        response = self.get_html_response(feconf.CREATOR_DASHBOARD_URL)
+        csrf_token = self.get_csrf_token_from_response(response)
+
+        suggestion_to_accept = self.get_json(
+            '%s?author_id=%s' % (
+                feconf.SUGGESTION_LIST_URL_PREFIX,
+                self.author_id))['suggestions'][0]
+
+        response = self.get_html_response(feconf.CREATOR_DASHBOARD_URL)
+        csrf_token = self.get_csrf_token_from_response(response)
+
+        with self.swap(constants, 'ENABLE_NEW_STRUCTURE_PLAYERS', True):
+            response = self.put_json(
+                '%s/topic/%s/%s' % (
+                    feconf.SUGGESTION_ACTION_URL_PREFIX,
+                    suggestion_to_accept['target_id'],
+                    suggestion_to_accept['suggestion_id']),
+                {'action': 'invalid_action'}, csrf_token=csrf_token,
+                expected_status_int=400)
+
+        self.assertEqual(
+            response['error'], 'Invalid action.')
+
+        self.logout()
+
+    def test_reject_suggestion(self):
```

Comment:
test_reject_suggestion_to_topic maybe? Are we not testing the "accept" action? Though probably without that test too, the coverage might be a 100%, I would suggest adding a test for that too.

Reply:
Done